navigator/frontend/src/applets/contract/Exercise.tsx

```diff
@@ -30,11 +30,11 @@ interface Local {
 }
 
 class Component extends React.Component<Props, Local> {
```

Comment:
We should think about refactoring this to using react.hooks as these are in my experience so far easier to understand and also don't need to make use of `UNSAFE_componentWillRecieveProps`.

Reply:
I know nothing about those.